src/database.js

```diff
@@ -0,0 +1,53 @@
+const dotenv = require('dotenv');
+const mongoose = require('mongoose');
+const Question = require('./models/leetCodeQuestion');
+dotenv.config();
+
+const connectToMongoDb = async () => {
+  const MONGO_CLIENT = process.env.ATLAS_URI || '';
+  await mongoose.connect(MONGO_CLIENT, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+  });
+
+  const mongoDb = mongoose.connection;
+  mongoDb.once('open', () => {
+    console.log('SUCCESS: Connected to the MongoDB database');
+  });
+  mongoDb.on('error', (error) => {
+    console.error('MongoDB database connection error:', error);
+  });
+};
+
+const closeMongoDbConnection = async () => {
+  await mongoose.connection.close();
+};
+
+/**
+ * Adds a new question if it doesn't exist, or updates an existing question.
+ * @param {Object} question - The question object to be added or updated.
+ * @throws {Error} If there was an error while adding/updating the question.
+ */
+const addOrUpdateQuestion = async (question) => {
+  try {
+    await Question.findOneAndUpdate(
+      { leetCodeId: question.leetCodeId },
```

Comment:
The program knows that the question is duplicate/not based on their id on leetcode?

Reply:
Yes. The first params of findOneAndUpdate searches for an existing document that has a match to leetCodeId, which is given by questioned of LeetCode API. If the leetCodeId don't exist, it will insert based on  the last params, upsert = true.